### ISSUE TITLE
Increase FPS by decreasing Render Resolution

### ISSUE THREAD

**Issue by djthegr8** [vladmandic/human#27]
How do I reduce the render resolution and resize the canvas into a smaller one (300 x 150)?

**Comment by vladmandic** [comment]
in the demo?  ou have slider "image height" and "image width" in the UI itself  

in the code?  
- for camera resolution, change `demo/browser.js`
```    video: {
      resizeMode: 'none',
      width: { ideal: 300 },
      height: { ideal: 150 },
    },
```
- for output resolution, change `demo/browser.js`  
```
const userConfig = { filter: { width: 300, height: 150 } }
```

**Comment by djthegr8** [comment]
Yet the canvas size remains same and I just get a blurred image

**Comment by vladmandic** [comment]
It's just CSS, actual canvas gets smaller.
Code in demo sets CSS style of canvas to to always be displayed at full screen width. Just remove that.
```
      canvas.style.width = canvas.width > canvas.height ? '100vw' : '';
      canvas.style.height = canvas.width > canvas.height ? '' : '100vh';
```

**Comment by djthegr8** [comment]
Thanks a ton

**Comment by djthegr8** [comment]
No significant FPS increase [here](https://human-demo-test.netlify.app/demo/)

**Comment by djthegr8** [comment]
Any other way?

**Comment by vladmandic** [comment]
@djthegr8 there is something in your implementation

your demo runs with 5FPS and mine with 10+FPS when I set it to run with 320p input resolution and WASM SIMD  
that's more than 100% difference  

when did you fork the code? i did a lot of optimizations in the last few days

**Comment by djthegr8** [comment]
I forked day before day before yesterday i think

**Comment by djthegr8** [comment]
And like I'm running just one model, this is tremendous difference

**Comment by djthegr8** [comment]
But, it may be because its using WASM instead of WebGL

**Comment by vladmandic** [comment]
my test above was with WASM SIMD, same as yours. huge difference.  
here's the screenshot running with WASM at 700x1080 at 6.3FPS average with all models enabled.  

![image](https://user-images.githubusercontent.com/57876960/98676192-48fb6b80-2329-11eb-95f9-949480702651.png)

**Comment by djthegr8** [comment]
Okay, so do I need to re-fork and then again do all the adjustments :weary:

**Comment by vladmandic** [comment]
you should be able to roll foward your branch for the library itself and keep your demo app.  
or copy your app to a separate folder, do a new fork and copy your demo app back.  

all of the optimizations are in the library, so your app should be fine except to remove <script> tags from html as they are no longer needed (wasm backend is now bundled).

**Comment by djthegr8** [comment]
Well, I actually edited the demo-browser-index.js thus I need to

**Comment by vladmandic** [comment]
never edit a compiled source. why not edit `browser.js`?  
you can use either use it directly or compile to `demo-browser-index.js`  

btw, only reason why i'm compiling demo in the first place is because mobile browsers don't yet implement <script type=module>, so can't use `browser.js` directly on mobile platforms.

**Comment by djthegr8** [comment]
Um alright

**Comment by djthegr8** [comment]
@vladmandic the library loads all models _irrespective_ of them being switched on or not...

**Comment by djthegr8** [comment]
Please fix this as this will lead to significant speed up

**Comment by djthegr8** [comment]
![image](https://user-images.githubusercontent.com/44142719/98680315-3dd72880-2387-11eb-86cb-9cfc86eb4854.png)

**Comment by djthegr8** [comment]
Also, it just keeps loading [here](https://human-demo-test.netlify.app/demo]

**Comment by vladmandic** [comment]
done

bug was only when using *async operations* (which is enabled by default) since it was loading everything in parallel instead of checking one-by-one what is needed  
note that this does NOT impact execution speed, only initial load speed  

`browser.js`:
```js
const userConfig = {};
```
default is to load all models:
```
Human: flags: {IS_BROWSER: true, IS_NODE: false, DEBUG: false, WEBGL_VERSION: 2, HAS_WEBGL: true}
age.js:15 Human: load model: models/age-ssrnet-imdb
gender.js:18 Human: load model: models/gender-ssrnet-imdb
blazeface.js:170 Human: load model: models/blazeface-back
emotion.js:18 Human: load model: models/emotion-large
facemesh.js:52 Human: load model: models/facemesh
facemesh.js:54 Human: load model: models/iris
modelPoseNet.js:44 Human: load model: models/posenet
handpose.js:79 Human: load model: models/handdetect
handpose.js:81 Human: load model: models/handskeleton
human.js:78 Human: warmed up
```

`browser.js`:
```js
const userConfig = {
  body: { enabled: false },
  hand: { enabled: false },
};
```
now it skips loading of body and hand:
```
Human: flags: {IS_BROWSER: true, IS_NODE: false, DEBUG: false, WEBGL_VERSION: 2, HAS_WEBGL: true}
gender.js:18 Human: load model: models/gender-ssrnet-imdb
age.js:15 Human: load model: models/age-ssrnet-imdb
emotion.js:18 Human: load model: models/emotion-large
blazeface.js:170 Human: load model: models/blazeface-back
facemesh.js:52 Human: load model: models/facemesh
facemesh.js:54 Human: load model: models/iris
human.js:78 Human: warmed up
```

**Comment by djthegr8** [comment]
Thanks for the fix 👍🏻
